pkg/k8sutil/status.go

```diff
@@ -43,6 +44,58 @@ func IsMarkedForDeletion(m metav1.ObjectMeta) bool {
 	return m.GetDeletionTimestamp() != nil
 }
 
+// UpdateBrokerConfigurationBackup updates the broker status with a backup from kafka broker configurations
+func UpdateBrokerConfigurationBackup(c client.Client, cluster *banzaicloudv1beta1.KafkaCluster) error {
+	needsUpdate, err := generateBrokerConfigurationBackups(cluster)
+	if err != nil {
+		return err
+	}
+	if !needsUpdate {
+		return nil
+	}
+	ctx := context.Background()
+	if err := c.Status().Update(ctx, cluster); err != nil {
+		if !apierrors.IsConflict(err) {
+			return errors.WrapIff(err, "could not update Kafka broker(s) configuration backup state")
+		}
+		if err := c.Get(ctx, types.NamespacedName{Namespace: cluster.Namespace, Name: cluster.Name}, cluster); err != nil {
+			return errors.WrapIf(err, "could not get config for updating status")
+		}
+		needsUpdate, err := generateBrokerConfigurationBackups(cluster)
+		if err != nil {
+			return err
+		}
+		if !needsUpdate {
+			return nil
+		}
+		if err = c.Status().Update(ctx, cluster); err != nil {
+			return errors.WrapIff(err, "could not update Kafka broker(s) configuration backup state")
+		}
+	}
+	return nil
+}
+
+func generateBrokerConfigurationBackups(cluster *banzaicloudv1beta1.KafkaCluster) (bool, error) {
+	needsUpdate := false
+	if cluster.Status.BrokersState == nil {
+		cluster.Status.BrokersState = make(map[string]banzaicloudv1beta1.BrokerState)
+	}
+
+	for _, broker := range cluster.Spec.Brokers {
+		brokerState := cluster.Status.BrokersState[fmt.Sprint(broker.Id)]
+		configurationBackup, err := util.GzipAndBase64BrokerConfiguration(&broker)
+		if err != nil {
+			return false, errors.WrapIfWithDetails(err, "could not generate broker configuration backup", "brokerId", broker.Id)
+		}
+		if !needsUpdate && configurationBackup != brokerState.ConfigurationBackup {
```

Comment:
1. needsUpdate is false, but if u see in the condition there is negation, so the first time it will true and later it will true so the negation will be false. This condition is only satisfied only once.

Reply:
Sorry, misread this part, you are right.